### ISSUE TITLE
Blunder: https://lichess.org/InxPeFUf

### ISSUE THREAD

**Issue by XInTheDark** [XInTheDark/PVplayer#16]
```
...
36.  Re4         -1.54     7/14       Nodes:   2.6 M     NPS:   1.0 M     MT: 00:02.5                       
Engine:  37.  Qd1         -1.64                Nodes:   1.9 M                      MT: 00:02.0                       
Engine:  38.  Rxf4        -0.67                Nodes: 802.7 k                      MT: 00:00.7                       
Engine:  39.  Qg4         -0.43     4/2        Nodes:   1.4 M     NPS: 340.3 k     MT: 00:01.1                       
Engine:  40.  Kc2         -0.29     13/15      Nodes:   3.3 M     NPS:   1.2 M     MT: 00:02.8                       
Engine:  41.  Qxc8+       -0.32     4/2        Nodes:   1.4 M     NPS: 341.9 k     MT: 00:01.1                       
Engine:  42.  Qe6         -0.26     12/21      Nodes:   4.5 M     NPS:   1.5 M     MT: 00:02.9                       
blundar_bot (player): Evaluation: 41... Kg7 +0.67 19/42 Nodes: 504.7 k NPS: 101.7 k MT: 00:05.0 Hash: 0.1 %
Engine:  43.  Qh3+        -0.27                Nodes: 942.3 k                      MT: 00:00.6                       
Engine:  44.  Qxh7        -0.27     11/27      Nodes:   3.5 M     NPS:   1.3 M     MT: 00:02.8                       
Engine:  45.  h4+         -0.29                Nodes:   1.2 M                      MT: 00:00.9                       
blundar_bot (player): Evaluation: 44... Rxd5 +0.80 16/28 Nodes: 398.6 k NPS: 100.5 k MT: 00:04.0 Hash: 0.4 %
Engine:  46.  Qxg6+       -0.29     9/27       Nodes:   3.4 M     NPS:   1.3 M     MT: 00:02.7                       
Engine:  47.  Qh7+        -0.29                Nodes: 943.0 k                      MT: 00:00.8                       
Engine:  48.  Qe7+        -0.25     12/24      Nodes:   3.6 M     NPS:   1.2 M     MT: 00:03.0                       
blundar_bot (player): Evaluation: 47... Rh5 +0.76 16/31 Nodes: 122.1 k NPS: 111.3 k MT: 00:01.1 Hash: 0.6 %
Engine:  49.  Nd4         -0.25                Nodes: 956.1 k                      MT: 00:00.7                       
Engine:  50.  Qe4         -0.22     9/26       Nodes:   3.9 M     NPS:   1.5 M     MT: 00:02.7                       
Engine:  51.  Qxf4+       -0.19                Nodes:   1.4 M                      MT: 00:01.0                       
Engine:  52.  Kb1         +0.13     11/18      Nodes:   2.9 M     NPS:   1.6 M     MT: 00:02.0                       
blundar_bot (player): Evaluation: 51... Kh5 +0.49 18/39 Nodes: 201.9 k NPS: 126.1 k MT: 00:01.6 Hash: 0.2 %
Engine:  53.  Kc1         -0.73     17/10      Nodes:   2.4 M     NPS:   1.0 M     MT: 00:02.4                       
blundar_bot (player): Evaluation: 52... b5 +1.11 17/27 Nodes: 453.2 k NPS: 99.5 k MT: 00:04.6 Hash: 0.5 %
Engine:  54.  Qe3         -1.59                Nodes: 869.7 k                      MT: 00:00.7                       
blundar_bot (player): Evaluation: 53... Kg6 +1.42 18/26 Nodes: 843.3 k NPS: 96.4 k MT: 00:08.7 Hash: 0.3 %
Engine:  55.  Qf4+        -1.77     6/15       Nodes:   4.2 M     NPS:   1.5 M     MT: 00:02.8                       
Engine:  56.  Qh4         -1.76                Nodes:   1.6 M                      MT: 00:01.0                       
blundar_bot (player): Evaluation: 55... Ke7 +1.92 15/25 Nodes: 369.7 k NPS: 104.1 k MT: 00:03.6 Hash: 0.5 %
Engine:  57.  Qh3         -1.72     7/14       Nodes:   4.0 M     NPS:   1.6 M     MT: 00:02.6                       
Engine:  58.  Kc2         -2.51                Nodes:   3.6 M                      MT: 00:02.0                       
Engine:  59.  Kb3         -2.84                Nodes:   2.0 M                      MT: 00:01.0                       
blundar_bot (player): Evaluation: 58... Qe4+ +1.76 27/39 Nodes: 1.2 M NPS: 161.3 k MT: 00:07.4 Hash: 0.3 %
Engine:  60.  Kc2         -2.87     12/245     Nodes:   4.0 M     NPS:   2.1 M     MT: 00:02.1                       
Engine:  61.  Qxg2        -3.15                Nodes:   3.3 M                      MT: 00:01.6                       
blundar_bot (player): Evaluation: 60... Qg2+ +1.47 27/35 Nodes: 419.9 k NPS: 169.4 k MT: 00:02.5 Hash: 0.3 %
Engine:  62.  Kc1         -2.90     38/13      Nodes:   3.6 M     NPS:   1.8 M     MT: 00:02.0                       
blundar_bot (player): Evaluation: 61... Rxg2+ +1.47 27/31 Nodes: 384.3 k NPS: 104.0 k MT: 00:03.7 Hash: 0.2 %
Engine:  63.  b3          -3.27                Nodes: 560.4 k                      MT: 00:00.3                       
Engine:  64.  Kb1         -2.98     26/14      Nodes:   5.0 M     NPS:   1.7 M     MT: 00:03.0                       
Engine:  65.  b4          -3.52                Nodes: 571.5 k                      MT: 00:00.3                       
blundar_bot (player): Evaluation: 64... Kd7 +2.89 29/35 Nodes: 898.9 k NPS: 153.7 k MT: 00:05.8 Hash: 0.5 %
Engine:  66.  Kb2         -2.97     16/16      Nodes:   6.7 M     NPS:   2.0 M     MT: 00:03.3                       
Engine:  67.  Kc2         -3.89                Nodes:   4.4 M                      MT: 00:02.1                       
Engine:  68.  Kc1         -3.71                Nodes: 507.7 k                      MT: 00:00.3                       
Engine:  69.  Kd2         -3.85     10/15      Nodes:   4.8 M     NPS:   1.6 M     MT: 00:03.0                       
```

**Comment by XInTheDark** [comment]
Also note 25. Kc1 is likely a decisive mistake as well.

**Comment by XInTheDark** [comment]
This was most likely fixed with 75a41d6 and possibly a few other patches that allowed very very shallow search. 
We need to think about how we can ensure the integrity of all evaluations, since the search gets ruined if one position is not evaluated properly. The current mechanism (recalculation & pruning) is not effective as it occurs once per few iterations only.
